frame/nfts/src/features/roles.rs

```diff
@@ -0,0 +1,47 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::*;
+use itertools::Itertools;
+
+impl<T: Config<I>, I: 'static> Pallet<T, I> {
```

Comment:
with multiple files the pallet looks more complex to me.
jumping to features/roles file, I wont expect impl of Pallet type, which initially defined in a different module (lib.rs file).
I also would expect here to see the Roles them-self, but we have them one level up in types.

It's all matter of taste (the way we structure it), but having many files, increases the entropy and raises more questions for those who reads it. Keeping it closer to the style we already have makes it simpler. 

can you tell me whats difference between impls in functions.rs and features/*?